src/pages/index.js

```diff
@@ -2,76 +2,226 @@ import "./index.css";
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      changePictureModal.renderLoading(false);
+      changePictureModal.close();
+    });
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  handleChangePicture
+);
+changePictureModal.setEventListeners();
+const pictureFormValidator = new FormValidator(config, pictureChangeForm);
+pictureFormValidator.enableValidation();
+profilePicture.addEventListener("click", () => {
+  console.log("not working");
+  pictureFormValidator.resetValidation();
+  changePictureModal.open();
+});
+
+const editProfileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  (userData) => {
+    editProfileModal.renderLoading(true);
+    console.log(userData);
+    api
+      .saveUserData(
+        { name: userData.title, about: userData.description },
+        userData.avatar
+      )
+      .then((res) => {
+        handleSetUserInfo(res);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        editProfileModal.renderLoading(false);
+      });
+    function handleSetUserInfo(userData) {
+      console.log(userData);
+      userInfo.setUserInfo(userData);
+    }
+  }
+);
+editProfileModal.setEventListeners();
+
+const previewImageModal = new PopupWithImage("#preview-image-modal");
+previewImageModal.setEventListeners();
+
+const addCardModal = new PopupWithForm("#profile-add-card", (cardData) => {
+  console.log(cardData);
+  addCardModal.renderLoading(true);
+  api
+    .addCard(cardData) // once
+    .then((res) => {
+      // here
+      renderCard(res);
+      // renderFunctionreplace it with a real function which renders card
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
+    })
+    .finally(() => {
+      addCardModal.renderLoading(false);
+    });
+});
+addCardModal.setEventListeners();
+
+function createCard(cardData) {
+  const newCard = new Card(
+    cardData,
+    cardTemplateSelector,
+    handleImageClick,
+    (card) => {
+      handleDeleteClick(card);
+    },
+
+    handleLikeClick
+  );
+  return newCard.getView();
 }
 
-function handleCardAddSubmit(data) {
-  console.log(data);
-  renderCard(data, cardListEl);
-  cardAddForm.reset();
+function handleImageClick(name, link) {
+  previewImageModal.open(name, link);
 }
 
 /* listeners */
 
+// deleteFormButton.addEventListener("click", () => {
+//   cardDeleteForm.open();
+// });
+
+// const cardDeleteForm = new PopupWithForm("#card-delete-modal");
+// cardDeleteForm.setEventListeners();
+
 profileEditButton.addEventListener("click", () => {
-  const useData = userInfo.getUserInfo();
-  nameInput.value = useData.name;
-  jobInput.value = useData.job;
-  editCardModal.open();
+  const values = userInfo.getUserInfo();
+  nameInput.value = values.name;
```

Comment:
**Could be improved**

Please, call `resetValidation` here as well to clear possible errors